### PR TITLE
first add fluidimage recipe

### PR DESCRIPTION
**Pull request by CyrilleBonamy**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by paugier** [comment]
I confirm!

File: recipes/fluidimage/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  number: 0
+  entry_points:
+    - fluidimviewer-pg = fluidimage.gui.pg_main:main
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+
+requirements:
+  host:
+    - fluiddyn >=0.2.0
+    - h5netcdf
+    - h5py
```
Comment (mariusvniekerk) [inline recipes/fluidimage/meta.yaml]: Are these dependencies needed at build time. Odds are you just need python pip and setuptools_scm

**Comment by paugier** [comment]
fluidimage 0.1.2 released on PyPI. it could solve the problem with Windows.

fluidimage now depends on transonic 0.2.2 (build and run times).

File: recipes/fluidimage/meta.yaml (hunk 1 above)
Comment (CyrilleBonamy) [reply recipes/fluidimage/meta.yaml]: I deleted all the unnecessary dependencies

**Comment by CyrilleBonamy** [comment]
> fluidimage 0.1.2 released on PyPI. it could solve the problem with Windows.
> 
> fluidimage now depends on transonic 0.2.2 (build and run times).

There are always problems with Windows. @paugier , do you have an idea?

TEST END: C:\Miniconda\conda-bld\win-64\fluidimage-0.1.2-py37_0.tar.bz2
Renaming work directory,  C:\Miniconda\conda-bld\fluidimage_1559774762593\work  to  C:\Miniconda\conda-bld\fluidimage_1559774762593\work_moved_fluidimage-0.1.2-py>=36_0_win-64_main_build_loop

Traceback (most recent call last):
  File "C:\Miniconda\lib\shutil.py", line 544, in move
    os.rename(src, real_dst)
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\Miniconda\\conda-bld\\fluidimage_1559774762593\\work' -> 'C:\\Miniconda\\conda-bld\\fluidimage_1559774762593\\work_moved_fluidimage-0.1.2-py>=36_0_win-64_main_build_loop'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".ci_support\build_all.py", line 131, in <module>
    build_all(args.recipes_dir, args.arch)
  File ".ci_support\build_all.py", line 68, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File ".ci_support\build_all.py", line 121, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "C:\Miniconda\lib\site-packages\conda_build\api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 2364, in build_tree
    shutil.move(metadata.config.work_dir, dest)
  File "C:\Miniconda\lib\shutil.py", line 555, in move
    symlinks=True)
  File "C:\Miniconda\lib\shutil.py", line 315, in copytree
    os.makedirs(dst)
  File "C:\Miniconda\lib\os.py", line 220, in makedirs
    mkdir(name, mode)
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\Miniconda\\conda-bld\\fluidimage_1559774762593\\work_moved_fluidimage-0.1.2-py>=36_0_win-64_main_build_loop'
##[error]Cmd.exe exited with code '1'.

**Comment by CyrilleBonamy** [comment]
> fluidimage 0.1.2 released on PyPI. it could solve the problem with Windows.
> 
> fluidimage now depends on transonic 0.2.2 (build and run times).

There are always problems with Windows. @paugier , do you have an idea?

TEST END: C:\Miniconda\conda-bld\win-64\fluidimage-0.1.2-py37_0.tar.bz2
Renaming work directory,  C:\Miniconda\conda-bld\fluidimage_1559774762593\work  to  C:\Miniconda\conda-bld\fluidimage_1559774762593\work_moved_fluidimage-0.1.2-py>=36_0_win-64_main_build_loop

Traceback (most recent call last):
  File "C:\Miniconda\lib\shutil.py", line 544, in move
    os.rename(src, real_dst)
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\Miniconda\\conda-bld\\fluidimage_1559774762593\\work' -> 'C:\\Miniconda\\conda-bld\\fluidimage_1559774762593\\work_moved_fluidimage-0.1.2-py>=36_0_win-64_main_build_loop'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".ci_support\build_all.py", line 131, in <module>
    build_all(args.recipes_dir, args.arch)
  File ".ci_support\build_all.py", line 68, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File ".ci_support\build_all.py", line 121, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "C:\Miniconda\lib\site-packages\conda_build\api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 2364, in build_tree
    shutil.move(metadata.config.work_dir, dest)
  File "C:\Miniconda\lib\shutil.py", line 555, in move
    symlinks=True)
  File "C:\Miniconda\lib\shutil.py", line 315, in copytree
    os.makedirs(dst)
  File "C:\Miniconda\lib\os.py", line 220, in makedirs
    mkdir(name, mode)
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\Miniconda\\conda-bld\\fluidimage_1559774762593\\work_moved_fluidimage-0.1.2-py>=36_0_win-64_main_build_loop'
##[error]Cmd.exe exited with code '1'.

**Comment by paugier** [comment]
@conda-forge-admin, please rerender

**Comment by paugier** [comment]
> There are always problems with Windows. @paugier , do you have an idea?

No and it does not seem to be a problem in fluidimage...

**Comment by paugier** [comment]
It seems that my `@conda-forge-admin, please rerender` did not work. Maybe you need to write it @CyrilleBonamy ?

**Comment by CyrilleBonamy** [comment]
@conda-forge-admin, please rerender

**Comment by paugier** [comment]
```
TEST END: C:\Miniconda\conda-bld\win-64\fluidimage-0.1.2-py37he980bc4_0.tar.bz2
Renaming work directory,  C:\Miniconda\conda-bld\fluidimage_1559816569987\work  to  C:\Miniconda\conda-bld\fluidimage_1559816569987\work_moved_fluidimage-0.1.2-py>=36he980bc4_0_win-64_main_build_loop

Traceback (most recent call last):
  File "C:\Miniconda\lib\shutil.py", line 544, in move
    os.rename(src, real_dst)
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\Miniconda\\conda-bld\\fluidimage_1559816569987\\work' -> 'C:\\Miniconda\\conda-bld\\fluidimage_1559816569987\\work_moved_fluidimage-0.1.2-py>=36he980bc4_0_win-64_main_build_loop'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".ci_support\build_all.py", line 131, in <module>
    build_all(args.recipes_dir, args.arch)
  File ".ci_support\build_all.py", line 68, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File ".ci_support\build_all.py", line 121, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "C:\Miniconda\lib\site-packages\conda_build\api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 2364, in build_tree
    shutil.move(metadata.config.work_dir, dest)
  File "C:\Miniconda\lib\shutil.py", line 555, in move
    symlinks=True)
  File "C:\Miniconda\lib\shutil.py", line 315, in copytree
    os.makedirs(dst)
  File "C:\Miniconda\lib\os.py", line 220, in makedirs
    mkdir(name, mode)
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\Miniconda\\conda-bld\\fluidimage_1559816569987\\work_moved_fluidimage-0.1.2-py>=36he980bc4_0_win-64_main_build_loop'
```

There is indeed a name problem. I guess the ">" in "work_moved_fluidimage-0.1.2-py>=36he980bc4_0_win-64_main_build_loop". 

But this name is created by conda-???, so I don't know how to solve this problem.

We could have a look at how other packages tell conda-forge to build only for python >= 3.6, without having this problem on Windows...

@conda-forge/staged-recipes 

- Any ideas how we can solve this problem?
- Why our `@conda-forge-admin, please rerender` have no effect here?

File: recipes/fluidimage/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: f2910a799bcd625a3a10ae6afb3e05668629e21764ea6ee171774a0817afee00
+
+build:
+  number: 0
+  entry_points:
+    - fluidimviewer-pg = fluidimage.gui.pg_main:main
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
```
Comment (chrisburr) [inline recipes/fluidimage/meta.yaml]:
```suggestion
  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
  skip: true  # [py>=36]
```

File: recipes/fluidimage/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - pip
+    - python >=3.6
```
Comment (chrisburr) [inline recipes/fluidimage/meta.yaml]:
```suggestion
    - python
```

File: recipes/fluidimage/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    - dask
+    - fluiddyn >=0.2.0
+    - h5netcdf
+    - h5py
+    - imageio
+    - matplotlib >=1.4.2
+    - numpy >=1.8
+    - pims
+    - pyfftw >=0.10.4
+    - python >=3.6
```
Comment (chrisburr) [inline recipes/fluidimage/meta.yaml]:
```suggestion
    - python
```

**Comment by chrisburr** [comment]
> Why our @conda-forge-admin, please rerender have no effect here?

It only applies to feedstocks. In `staged-recipes` a customised CI configuration is used.

> We could have a look at how other packages tell conda-forge to build only for python >= 3.6, without having this problem on Windows...

That style of constraining Python should only be used for `noarch: python` packages. When `noarch` can't be used `skip` is the correct way of doing it (see my code comments). This might fix the issue.

File: recipes/fluidimage/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: f2910a799bcd625a3a10ae6afb3e05668629e21764ea6ee171774a0817afee00
+
+build:
+  number: 0
+  entry_points:
+    - fluidimviewer-pg = fluidimage.gui.pg_main:main
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+  skip: true  # [py>=36]
```
Comment (chrisburr) [inline recipes/fluidimage/meta.yaml]: Sorry I made a mistake, this should have been:
```suggestion
  skip: true  # [py<36]
```

**Comment by CyrilleBonamy** [comment]
@mariusvniekerk can you confirm my changes are ok for you? :)

**Comment by CyrilleBonamy** [comment]
And thank you @chrisburr for your help :-)

File: recipes/fluidimage/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    - imageio
+    - matplotlib >=1.4.2
+    - numpy >=1.8
+    - pims
+    - pyfftw >=0.10.4
+    - python
+    - scipy >=0.14.1
+    - trio
+    - ipython
+    - setuptools_scm
```
Comment (chrisburr) [inline recipes/fluidimage/meta.yaml]: Do you need this at runtime?
```suggestion
```

File: recipes/fluidimage/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    - transonic
+    - pythran
+  run:
+    - dask
+    - fluiddyn >=0.2.0
+    - h5netcdf
+    - h5py
+    - imageio
+    - matplotlib >=1.4.2
+    - numpy >=1.8
```
Comment (chrisburr) [inline recipes/fluidimage/meta.yaml]: Are you using the NumPy C API? If you are you should add `numpy` to `host` and change this to:
```suggestion
    - {{ pin_compatible('numpy') }}
```

File: recipes/fluidimage/meta.yaml (hunk 2 above)
Comment (chrisburr) [inline recipes/fluidimage/meta.yaml]: This can be simplified:
```suggestion
  script: "{{ PYTHON }} -m pip install . -vv "
```

**Comment by CyrilleBonamy** [comment]
@chrisburr can you merge ? ;)

**Comment by CyrilleBonamy** [comment]
@chrisburr is it ok for you now? ;)

**Comment by chrisburr** [comment]
Yup, thanks for fixing it 😃

**Pull request merged**